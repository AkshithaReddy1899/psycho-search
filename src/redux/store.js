import {
  legacy_createStore as createStore,
  combineReducers,
  applyMiddleware,
  compose,
} from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import personsReducer from './person';

const rootReducer = combineReducers({
  person: personsReducer,
});

const store = createStore(rootReducer, compose(applyMiddleware(logger, thunk)));

export default store;
