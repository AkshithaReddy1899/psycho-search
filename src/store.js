import { configureStore } from '@reduxjs/toolkit';
import personSlice from './slices/personSlice';

const reducer = {
  person: personSlice,
};

const store = configureStore({
  reducer,
  devTools: true,
  middleware: (getDefaultMiddleware) => getDefaultMiddleware({
    serializableCheck: false,
  }),
});

export default store;
