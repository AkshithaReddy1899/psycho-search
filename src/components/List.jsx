import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import { getDataFromApi } from '../redux/person';

function List() {
  const dispatch = useDispatch();
  const storeData = useSelector((store) => store.person);

  useEffect(() => {
    dispatch(getDataFromApi());
  }, []);

  const personList = [];

  if (storeData.length > 0) {
    storeData.forEach((obj) => {
      const data = obj.data.person;
      data.forEach((item) => {
        personList.push(item);
      });
    });
  }

  return (
    <div>
      <ul>
        {
					(personList.length) !== 0 ? (
					  personList.map((item) => (
  <Link to="/person" key={item.id} state={{ id: item.id }}>
    <li key={item.id} className="container border">
      <img src={item.image} alt={item.name} />
      <h3>{item.name}</h3>
      <p>
        <b>Crime: </b>
        {item.crime}
      </p>
      <p>
        <b>Region: </b>
        {item.country}
        ,&nbsp;
        {item.state}
      </p>
      <p>
        <b>Victims: </b>
        {item.victims}
      </p>
      <p>
        <b>Died at the age of: </b>
        {item.age_of_death}
      </p>
      <p>
        <b>Death: </b>
        {item.death}
      </p>
    </li>
  </Link>
					  ))) : (
  <div className="center">Oops! Something went wrong try again later</div>)
}
      </ul>
    </div>
  );
}

export default List;
