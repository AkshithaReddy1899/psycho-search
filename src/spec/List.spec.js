import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import matches from './test-utils';
import store from '../redux/store';
import List from '../components/List';

describe('Home', () => {
  it('Home matched snapshot', () => {
    matches(<Provider store={store}><BrowserRouter><List /></BrowserRouter></Provider>);
  });
});
