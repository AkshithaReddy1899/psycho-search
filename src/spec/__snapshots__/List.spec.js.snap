// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Home matched snapshot 1`] = `
<div>
  <ul
    className="flex flex-wrap flex-row justify-around align-center m-8"
  >
    <div
      className="center p-20"
    >
      <span
        style={
          Object {
            "display": "inherit",
          }
        }
      >
        <span
          style={
            Object {
              "animation": "react-spinners-RiseLoader-odd 1s 0s infinite cubic-bezier(0.15, 0.46, 0.9, 0.6)",
              "animationFillMode": "both",
              "backgroundColor": "white",
              "borderRadius": "100%",
              "display": "inline-block",
              "height": "25px",
              "margin": "2px",
              "width": "25px",
            }
          }
        />
        <span
          style={
            Object {
              "animation": "react-spinners-RiseLoader-even 1s 0s infinite cubic-bezier(0.15, 0.46, 0.9, 0.6)",
              "animationFillMode": "both",
              "backgroundColor": "white",
              "borderRadius": "100%",
              "display": "inline-block",
              "height": "25px",
              "margin": "2px",
              "width": "25px",
            }
          }
        />
        <span
          style={
            Object {
              "animation": "react-spinners-RiseLoader-odd 1s 0s infinite cubic-bezier(0.15, 0.46, 0.9, 0.6)",
              "animationFillMode": "both",
              "backgroundColor": "white",
              "borderRadius": "100%",
              "display": "inline-block",
              "height": "25px",
              "margin": "2px",
              "width": "25px",
            }
          }
        />
        <span
          style={
            Object {
              "animation": "react-spinners-RiseLoader-even 1s 0s infinite cubic-bezier(0.15, 0.46, 0.9, 0.6)",
              "animationFillMode": "both",
              "backgroundColor": "white",
              "borderRadius": "100%",
              "display": "inline-block",
              "height": "25px",
              "margin": "2px",
              "width": "25px",
            }
          }
        />
        <span
          style={
            Object {
              "animation": "react-spinners-RiseLoader-odd 1s 0s infinite cubic-bezier(0.15, 0.46, 0.9, 0.6)",
              "animationFillMode": "both",
              "backgroundColor": "white",
              "borderRadius": "100%",
              "display": "inline-block",
              "height": "25px",
              "margin": "2px",
              "width": "25px",
            }
          }
        />
      </span>
    </div>
  </ul>
</div>
`;
